packages/auth-client/test/index.js

```diff
@@ -0,0 +1,108 @@
+const chai = require('chai')
+const expect = chai.expect
+const nock = require('nock')
+const Eth = require('web3-eth')
+
+const stringify = require('json-stable-stringify')
+
+global.fetch = require('cross-fetch')
+
+import AuthClient from '../src/auth-client'
+
+const Web3Eth = new Eth()
+
+const USER_ADDRESS = '0x627306090abaB3A6e1400e9345bC60c78a8BEf57'
+const USER_PRIVATE_KEY =
+  '0xC87509A1C067BBDE78BEB793E6FA76530B6382A4C0241E5E4A9EC0A0F44DC0D3'
+const USER_ACCOUNT = Web3Eth.accounts.privateKeyToAccount(USER_PRIVATE_KEY)
+
+const AUTH_SERVER_HOST = 'https://auth.originprotocol.com'
+
+const AUTH_MESSAGE = 'SOME_AUTH_MESSAGE'
+
+const signAuthMessage = (timestamp = Date.now()) => {
+  const payload = stringify({
+    message: AUTH_MESSAGE,
+    timestamp
+  })
+
+  return {
+    signature: USER_ACCOUNT.sign(payload).signature,
+    payload: JSON.parse(payload),
+    timestamp
+  }
+}
+
+describe('Auth Client', () => {
+  it('should create an instance', () => {
+    const client = new AuthClient({
+      authServer: AUTH_SERVER_HOST,
+      activeWallet: USER_ADDRESS,
+      disablePersistence: true
+    })
+
+    expect(client.getActiveWallet()).to.equal(USER_ADDRESS)
+  })
+
+  it('should allow active wallet to be changed', () => {
+    const client = new AuthClient({
+      authServer: AUTH_SERVER_HOST,
+      activeWallet: USER_ADDRESS,
+      disablePersistence: true
+    })
+
+    expect(client.getActiveWallet()).to.equal(USER_ADDRESS)
+    client.setActiveWallet('0x0000000000000000000000000000000000000000')
+    expect(client.getActiveWallet()).to.equal(
+      '0x0000000000000000000000000000000000000000'
+    )
+  })
+
+  it('should generate a token', async () => {
+    nock(AUTH_SERVER_HOST)
+      .post('/api/tokens')
+      .reply(201, {
+        success: true,
+        authToken: 'Hello Token'
+      })
+
+    const client = new AuthClient({
+      authServer: AUTH_SERVER_HOST,
+      activeWallet: USER_ADDRESS,
+      disablePersistence: true
+    })
+
+    const { signature, payload } = signAuthMessage()
+
+    const { authToken } = await client.getTokenWithSignature(signature, payload)
```

Comment:
This test doesn't seem to validate that the client actually sent a well formed request, just that it sent some kind of request to the right URL.

Reply:
Thanks for catching this. Will update them

Reply:
Did this get added? Still looks like it doesn't actually test one of the format of the message to the server, or an actual request to the server.

Reply:
Yeah, I added the body contents to the `nock` server above. This should verify that the exact contents are sent by the client. If the `auth-client` doesn't send a request to that URL, nock wouldn't capture it and the test would fail.

https://github.com/OriginProtocol/origin/pull/3807/files#diff-6b05ef2644bfa77cb6cb5803238636aaR64-R73

Seems good to me. Or did I overlook something?